internal/controller/utils.go

```diff
@@ -72,3 +109,17 @@ func hashBytes(sourceStr []byte) (string, error) {
 	}
 	return fmt.Sprintf("%x", hashFunc.Sum(nil)), nil
 }
+
+func getSecretContent(rclient client.Client, secretName string, namespace string, secretField string) (string, error) {
+	foundSecret := &corev1.Secret{}
+	ctx := context.Background()
+	err := rclient.Get(ctx, client.ObjectKey{Name: secretName, Namespace: namespace}, foundSecret)
+	if err != nil {
+		return "", fmt.Errorf("Error fetching secret: %w", err)
```

Comment:
Can we add the Secret name part of the log ?

Reply:
Added secret name as well.